### PR TITLE
Bail out for benchmark run failures

### PR DESCRIPTION
**Pull request by gdvalle**
Exit with non-zero return code during benchmark failures.

### CODE REVIEW

File: lib/Test/Nginx/Socket.pm [hunk 1]
```diff
@@ -611,7 +611,11 @@ sub run_test_helper ($$) {
         $cmd = quote_sh_args($cmd);
 
         warn "$cmd\n";
-        system "unbuffer $cmd > /dev/stderr";
+        my $full_cmd = "unbuffer $cmd > /dev/stderr";
```
Comment (agentzh) [inline lib/Test/Nginx/Socket.pm]: Better reuse the existing `$cmd` variable?

File: lib/Test/Nginx/Socket.pm [hunk 2]
```diff
@@ -611,7 +611,11 @@ sub run_test_helper ($$) {
...
+        my $rc = system $full_cmd;
+        if ($rc != 0) {
```
Comment (agentzh) [inline lib/Test/Nginx/Socket.pm]: Better use the more compact form `if (system($cmd) != 0)` here instead.

File: lib/Test/Nginx/Socket.pm [hunk 3]
```diff
@@ -611,7 +611,11 @@ sub run_test_helper ($$) {
...
+          bail_out("Failure running '${full_cmd}': return code ${rc}");
```
Comment (agentzh) [inline lib/Test/Nginx/Socket.pm]: You can use the special variable `$?` here for the error code. Also, better to use "Failed to run '...'" as the error message. And the curly braces for the `$rc` reference is not necessary.

**Comment by gdvalle** [comment]
:+1: Thanks for the fast review @agentzh! I liked your suggestions and implemented each.

**Pull request closed without merging**